### PR TITLE
i#5427 compress: Turn -raw_compress lz4 on by default

### PR DESCRIPTION
**Pull request by derekbruening**
For non-static drmemtrace, -raw_compress is now set to lz4 by default,
as that is nearly always faster than no compression or any other
compression scheme we've tried.

If lz4 is not enabled in the build, and again drmemtrace is not
static, snappy_nocrc is the default.

Otherwise, compression is disabled by default, just like today.

Inverts the raw-lz4 tests to become raw-none tests.

Updates the documentation.

Issue: #5427

### CODE REVIEW

File: suite/tests/CMakeLists.txt [hunk 1]
```diff
@@ -3585,8 +3585,9 @@ if (BUILD_CLIENTS)
       set(tool.drcacheoff.raw-gzip_expectbase "offline-simple")
     endif ()
     if (liblz4)
```
Comment (abhinav92003) [inline suite/tests/CMakeLists.txt:3587]: If we're testing no compression, we don't need the liblz4 guard.

File: suite/tests/CMakeLists.txt [hunk 2]
```diff
@@ -3657,10 +3658,11 @@ if (BUILD_CLIENTS)
       set(tool.drcacheoff.windows-gzip_expectbase "offline-windows-simple")
     endif ()
     if (liblz4)
```
Comment (abhinav92003) [inline suite/tests/CMakeLists.txt:3660]: Same here.

**Review by abhinav92003 — APPROVED** [review]

File: suite/tests/CMakeLists.txt (hunk 1 above)
Comment (derekbruening) [reply suite/tests/CMakeLists.txt:3587]: Done.

File: suite/tests/CMakeLists.txt (hunk 2 above)
Comment (derekbruening) [reply suite/tests/CMakeLists.txt:3660]: Done.

**Pull request merged**